lib/widgets/messages.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:chat_app/widgets/chat_bubble.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class Messages extends StatelessWidget {
+  const Messages({super.key});
+
+  bool _hasPrevContinuousMessage(
+    List<QueryDocumentSnapshot<Map<String, dynamic>>> chatDocs,
+    int index,
+  ) {
+    if (index == chatDocs.length - 1) {
+      return false;
+    }
+    return chatDocs[index]['userID'] == chatDocs[index + 1]['userID'] &&
+        chatDocs[index]['time'].toDate().year ==
+            chatDocs[index + 1]['time'].toDate().year &&
+        chatDocs[index]['time'].toDate().month ==
+            chatDocs[index + 1]['time'].toDate().month &&
+        chatDocs[index]['time'].toDate().day ==
+            chatDocs[index + 1]['time'].toDate().day &&
+        chatDocs[index]['time'].toDate().hour ==
+            chatDocs[index + 1]['time'].toDate().hour &&
+        chatDocs[index]['time'].toDate().minute ==
+            chatDocs[index + 1]['time'].toDate().minute;
+  }
```

Comment:
```suggestion
bool _hasPrevContinuousMessage(
  List<QueryDocumentSnapshot<Map<String, dynamic>>> chatDocs,
  int index,
) {
  if (index == chatDocs.length - 1) return false;

  final QueryDocumentSnapshot currentDoc = chatDocs[index];
  final QueryDocumentSnapshot nextDoc = chatDocs[index + 1];

  final bool isSameUser = currentDoc['userID'] == nextDoc['userID'];
  final bool isSameTime = currentDoc['time'].toDate().difference(nextDoc['time'].toDate()).inMinutes == 0;

  return isSameUser && isSameTime;
}
```


밑의 두 함수도 이거처럼 수정되면 더 좋지 않을까요
toDate() 같은 함수를 불필요하게 여러 번 호출하는 건 퍼포먼스에 좋지 않아보입니다